src/org/labkey/test/params/experiment/DataClassDefinition.java

```diff
@@ -0,0 +1,173 @@
+package org.labkey.test.params.experiment;
+
+import org.jetbrains.annotations.NotNull;
+import org.labkey.remoteapi.domain.Domain;
+import org.labkey.remoteapi.domain.PropertyDescriptor;
+import org.labkey.test.params.FieldDefinition;
+import org.labkey.test.params.property.DomainProps;
+import org.labkey.test.util.TestDataGenerator;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Defines a Data Class/Data Type. Suitable for use with UI and API helpers.
+ * 'exp.datas'
+ */
+public class DataClassDefinition extends DomainProps
+{
+    private String _name;
+    private String _nameExpression;
+    private String _description;
+    private String _category;
+    private String _materialSource;
+    private List<FieldDefinition> _fields = new ArrayList<>();
+
+    public DataClassDefinition() { }
```

Comment:
Are you sure you want to keep this constructor? Having this constructor, and a setter for the name property will allow _name to be null which is never checked in getOptions.